nav2_planner/src/planner_server.cpp

```diff
@@ -184,25 +184,32 @@ PlannerServer::computePathToPose()
       goal = action_server_->accept_pending_goal();
     }
 
-    RCLCPP_DEBUG(get_logger(), "Attempting to a find path from (%.2f, %.2f) to "
+    RCLCPP_INFO(get_logger(), "Attempting to a find path from (%.2f, %.2f) to "
       "(%.2f, %.2f).", start.pose.position.x, start.pose.position.y,
-      goal->pose.pose.position.x, goal->pose.pose.position.y);
-
-    result->path = planner_->createPlan(start, goal->pose);
+      goal->poses.poses.back().pose.position.x, goal->poses.poses.back().pose.position.y);
+
+    result->path.poses.clear();
+    result->path.poses.push_back(start);
+    for (auto current_waypoint : goal->poses.poses)
+    {
+      auto partial_path = planner_->createPlan(result->path.poses.back(), current_waypoint);
```

Comment:
I'm not sure if this would fail immediately if a path to the `current_waypoint` is not possible or continue the loop for the next waypoints.